src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action in actions) {
+    const point = actions[action];
+
+    if (point.type === 'addProperties') {
+      Object.assign(state, point.extraData);
+    }
+
+    if (point.type === 'removeProperties') {
+      for (let i = 0; i < point.keysToRemove.length; i++) {
+        delete state[point.keysToRemove[i]];
+      }
+    }
+
+    if (point.type === 'clear') {
+      for (const n in state) {
```

Comment:
```suggestion
      for (const key in state) {
```

don't forget to use more descriptive variables names